core/domain/exp_services_test.py

```diff
@@ -881,13 +881,132 @@ def test_export_by_versions(self):
 def _get_change_list(state_name, property_name, new_value):
     """Generates a change list for a single state change."""
     return [{
-        'cmd': 'edit_state_property',
+        'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
         'state_name': state_name,
         'property_name': property_name,
         'new_value': new_value
     }]
 
 
+class UpdateGadgetTests(ExplorationServicesUnitTests):
+    """Test updating a gadget."""
+
+    def setUp(self):
+        super(UpdateGadgetTests, self).setUp()
+        exploration = self.save_new_valid_exploration(
+            self.EXP_ID, self.owner_id)
+        # Default outcome specification for an interaction.
+        self.init_state_name = exploration.init_state_name
+
+        self.interaction_default_outcome = {
+            'dest': self.init_state_name,
+            'feedback': ['Incorrect', '<b>Wrong answer</b>'],
+            'param_changes': []
+        }
+
+    def test_add_gadget_cmd(self):
+        """test adding of a gadget"""
```

Comment:
I don't think we do this, actually? It's the first time I've heard of it, and the other files don't seem to be doing this either...